lib/phoenix_live_view/test/client_proxy.ex

```diff
@@ -313,58 +313,94 @@ defmodule Phoenix.LiveViewTest.ClientProxy do
                :ok <- maybe_enabled(type, node, element),
                {:ok, event_or_js} <- maybe_event(type, node, element),
                {:ok, extra} <- maybe_values(type, node, element) do
-            {event, js_values, js_target_selector} = maybe_js_event(event_or_js)
-            extra = Map.merge(extra, js_values)
-
-            {values, uploads} =
-              case value do
-                %Upload{} = upload -> {extra, upload}
-                other -> {DOM.deep_merge(extra, stringify(other, & &1)), nil}
-              end
-
-            js_targets = DOM.targets_from_selector(root, js_target_selector)
-            node_targets = DOM.targets_from_node(root, node)
-
-            targets =
-              case {js_targets, node_targets} do
-                {[nil], right} -> right
-                {left, [nil]} -> left
-                {left, right} -> Enum.uniq(left ++ right)
-              end
-
-            {view, targets, event, values, uploads}
+            # There may be multiple `push` events collected from JS commands,
+            # but we take care to only send the `value` and any `extra` values
+            # (collected from phx-value or form inputs) once with the first event.
+            #
+            # > I can't quite reckon when you might be able to send phx-value-x values
+            # > and uploads from a phx-click and also trigger JS events, but this
+            # > method retains any current behaviour, even if aberrant.
```

Comment:
@rktjmp the upload comes an external parameter but I think it is reasonable to not allow it on JS.push. Can you please check the `phx-value` one? I am almost sure we don't include it in JS.push. If that's the case for both, I think we can move the values/upload handling to inside `maybe_js_events` (i.e. if it is a JS event, we don't attached it, otherwise we do).

Reply:
I went back to check and currently in 0.18.18, the additional values _are_ pulled in, eg:

```
    <div
      phx-click={
        JS.push("an-event")
        |> JS.push("an-event", value: %{"js" => true})
      }
      phx-value-key="something">
      button
    </div>
```

Gives

```
handle an-event: %{"key" => "something"}
handle an-event: %{"js" => true, "key" => "something"}
```

So I updated it to always send all values along and left the map(s) merging in the main block.

I left the uploads attached as I'm not sure if there are edge cases where it would be included.